xclim/indices/_threshold.py

```diff
@@ -1431,16 +1503,143 @@ def days_with_snow(
+        `const` is only used if `snr` is None.
+        Use this option if unit of low and/or high is mm/day
+        to convert prsn [kg m-2 s-1] to [mm day-1].
+    freq : str
+        Resampling frequency defining the periods as defined in
+        https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#resampling.
+
+    Returns
+    -------
+    xarray.DataArray, [%]
+        Percentage of days where snowfall exceeds a threshold.
+
+    Notes
+    -----
+        The estimated mean snow density value of 312 kg m-3 is taken from :cite:t:`sturm_swe_2010`.
+
+    References
+    ----------
+    :cite:cts:`frei_snowfall_2018`
+    """
+    sd = days_with_snow(prsn, low=thresh, snr=snr, const=const, freq=freq)
+    ndays = prsn.resample(time=freq).count(dim="time")
+    sfreq = sd / ndays * 100
+    sfreq = sfreq.assign_attr(**sd.attrs)
+    sfreq.attrs["units"] = "%"
+    return sfreq
+
+
+@declare_units(
+    prsn="[precipitation]",
+    thresh="[precipitation]",
+    snr="[mass]/[volume]",
+    const="[mass]/[volume]",
+)
+def snowfall_intensity(
+    prsn: xarray.DataArray,
+    thresh: Quantified = "1 mm/day",
+    snr: xarray.DataArray | None = None,
+    const: Quantified = "312 kg m-3",
+    freq: str = "AS-JUL",
+) -> xarray.DataArray:
+    r"""Days with snow.
+
+    Return mean daily solid precipitation during days with solid precipitation exceeds a threshold (default: 1 mm/day)
+
+    Parameters
+    ----------
+    prsn : xarray.DataArray
+        Solid precipitation flux.
+    thresh : Quantified
+        Threshold precipitation flux on which to base evaluation.
+    snr : xarray.DataArray, optional
+        Snow density [kg/m^3].
+        Use this option if unit of low and/or high is mm/day
+        to convert prsn [kg m-2 s-1] to [mm day-1].
+        If `snr` is None `const` will be used for convertion.
+    const: Quantified
+        Constant snow density [kg/m^3]
+        `const` is only used if `snr` is None.
+        Use this option if unit of low and/or high is mm/day
+        to convert prsn [kg m-2 s-1] to [mm day-1].
+    freq : str
+        Resampling frequency defining the periods as defined in
+        https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#resampling.
+
+    Returns
+    -------
+    xarray.DataArray, [%]
+        Mean daily snowfall during days where snowfall exceeds a threshold.
+
+    Notes
+    -----
+        The estimated mean snow density value of 312 kg m-3 is taken from :cite:t:`sturm_swe_2010`.
+
+    References
+    ----------
+    :cite:cts:`frei_snowfall_2018`
+    """
+    unit = str2pint("mm/day")
+    unit_thresh = str2pint(thresh)
+    if unit_thresh._units == unit._units:
+        prsn = prsn_to_mm_per_day(prsn, snr=snr, const=const)
+    thresh = convert_units_to(thresh, prsn, context="hydro")
+    cond = prsn >= thresh
+    mean = cond.resample(time=freq).mean(dim="time")
+    return xarray.where(mean == np.nan, 0, mean)
```

Comment:
I think that this current function `snowfall_intensity` achieves what is expected from `snowfall_frequency` (you would only need an extra factor of 100), since you sum on the condition.
 
To get the intensity, you could mask  `prsn` and then sum on it
```suggestion
    cond = prsn >= thresh
    mean = prsn.where(cond).resample(time=freq).mean(dim="time")
    return mean.where(mean.notnull(), 0)
```